src/main/java/com/cleanroommc/groovyscript/compat/mods/avaritia/Compressor.java

```diff
@@ -0,0 +1,115 @@
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.ingredient.IngredientHelper;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import morph.avaritia.recipe.AvaritiaRecipeManager;
+import morph.avaritia.recipe.compressor.CompressorRecipe;
+import morph.avaritia.recipe.compressor.ICompressorRecipe;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+
+public class Compressor extends VirtualizedRegistry<ICompressorRecipe> {
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(recipe -> AvaritiaRecipeManager.COMPRESSOR_RECIPES.values().removeIf(r -> r == recipe));
+        restoreFromBackup().forEach(recipe -> AvaritiaRecipeManager.COMPRESSOR_RECIPES.put(recipe.getRegistryName(), recipe));
+    }
+
+    public boolean remove(ICompressorRecipe recipe) {
+        recipe = AvaritiaRecipeManager.COMPRESSOR_RECIPES.remove(recipe.getRegistryName());
+        if (recipe != null) {
+            addBackup(recipe);
+            return true;
+        }
+        return false;
+    }
+
+    public boolean removeByOutput(ItemStack output) {
+        if (IngredientHelper.isEmpty(output)) {
+            GroovyLog.msg("Error removing avaritia compressor recipe")
+                    .add("output must not be empty")
+                    .error()
+                    .post();
+            return false;
+        }
+        return AvaritiaRecipeManager.COMPRESSOR_RECIPES.values().removeIf(recipe -> {
+            if (recipe != null && recipe.getResult().isItemEqual(output)) {
+                addBackup(recipe);
+                return true;
+            }
+            return false;
+        });
+    }
+
+    public void removeAll() {
+        AvaritiaRecipeManager.COMPRESSOR_RECIPES.values().forEach(this::addBackup);
+        AvaritiaRecipeManager.COMPRESSOR_RECIPES.values().clear();
+    }
+
+    public SimpleObjectStream<ICompressorRecipe> streamRecipes() {
+        return new SimpleObjectStream<>(AvaritiaRecipeManager.COMPRESSOR_RECIPES.values()).setRemover(this::remove);
+    }
+
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    public void add(ICompressorRecipe recipe) {
+        AvaritiaRecipeManager.COMPRESSOR_RECIPES.put(recipe.getRegistryName(), recipe);
+        addScripted(recipe);
+    }
+
+    public void add(ItemStack output, IIngredient input, int cost) {
+        recipeBuilder()
+                .cost(cost)
+                .input(input)
+                .output(output)
+                .register();
+    }
+
+    public class RecipeBuilder extends AbstractRecipeBuilder<ICompressorRecipe> {
+
+        private int cost = 300;
+
+        public RecipeBuilder cost(int cost) {
```

Comment:
for `ExtendedCrafting.CompressionCrafting.RecipeBuilder` i added another way to get the item amount - consider adding the same here?

Reply:
this might have been a bad idea, actually - since now, if `inputCount(100).input(item)` the inputCount will be overridden to be 1, making the output of the builder dependent on the order elements are built - definitely undesired.

Reply:
How about only setting input count if ingredient amount is > 1

Reply:
a reasonable solution. would you mind also applying this to `CompressionCrafting.RecipeBuilder#input`?